plugins/catalog-graph/src/components/EntityRelationsGraph/types.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DependencyGraphTypes } from '@backstage/core-components';
+import { MouseEventHandler } from 'react';
+
+/**
+ * Edge between two entities.
+ *
+ * @public
+ */
+export type EntityEdge = DependencyGraphTypes.DependencyEdge<{
+  /**
+   * Up to two relations that are connecting an entity.
+   */
+  relations: string[];
+  /**
+   * Whether the entity is visible or not.
+   */
+  // Not used, but has to be non empty to draw a label at all!
+  label: 'visible';
+}>;
+
+/**
+ * Node representing an entity.
+ *
+ * @public
+ */
+export type EntityNode = DependencyGraphTypes.DependencyNode<{
+  /**
+   * Name of the entity.
+   */
+  name: string;
+  /**
+   * Optional kind of the entity.
+   */
+  kind?: string;
```

Comment:
would `type?: string` fit well in here too?

Reply:
especially if that together with kind could be used to draw custom icons in the future

Reply:
It's not used right now, but yeah, could be added later.

Reply:
We could also pass the full entity in and make the nodes more customizable. But that is maybe also something for later.